endpoints/openrtb2/auction_test.go

```diff
@@ -227,6 +228,54 @@ func TestBlacklistRequests(t *testing.T) {
 	tests.assert(t)
 }
 
+// TestRejectAccountRequired asserts we return a 400 code on a request that comes with no user id nor app id
+// if the `AccountRequired` field in the `config.Configuration` structure is set to true
+func TestRejectAccountRequired(t *testing.T) {
+	tests := &getResponseFromDirectory{
```

Comment:
Why not just add this test request to `endpoints/openrtb2/sample-requests`  and then create a new `getResponseFromDirectory` for those tests?

Would also suggest adding some more test cases like when account ID is not required and not provided in the request, required and provided, required and provided but blacklisted, etc.

Reply:
Four new test cases where added under `TestRejectAccountRequired` in `endpoints/openrtb2/auction_test.go`. With said additions, we are now covering all the blow scenarios
```
   | required | provided | blacklisted ||   expectedCode            |  test
---------------------------------------------------------------------------
 1 |   FALSE  |   FALSE  |   FALSE     || StatusOK                  |  8.1
 2 |   FALSE  |   FALSE  |    TRUE     || StatusOK                  |  8.1
 3 |   FALSE  |    TRUE  |   FALSE     || StatusOK                  |  TestGoodRequest
 4 |   FALSE  |    TRUE  |    TRUE     || StatusBadRequest          |  TestBlacklistRequests
 5 |    TRUE  |   FALSE  |   FALSE     || StatusOK                  |  8.2
 6 |    TRUE  |   FALSE  |    TRUE     || StatusOK                  |  8.2
 7 |    TRUE  |    TRUE  |   FALSE     || StatusOK                  |  8.3
 8 |    TRUE  |    TRUE  |    TRUE     || StatusServiceUnavailable  |  8.4
```